File: pset1/mario/mario.c
/**
 * Implements "Mario more comfortable" assignment.
 * 
 * Creates "super Mario's style" pyramids using a user definged number of '#' as bricks.
 * 
 * */

#include <stdio.h>
#include <cs50.h>

int main(void)
{
  //ask user for pyramid heigth. must be an integer between 1 and 8 inclusive.
  int height = 0;
  do
  {
    height = get_int("Height: ");
  } 
  while (height < 1 || height > 8);

  //print each line of the pyramid
  int line = 1;
  while (line <= height)
  {
    //print spaces
    for (int i = 0; i < height - line; i++)
    {
      printf(" ");
    }
    //print right side hashes
    for (int j = 0; j < line; j++)
    {
      printf("#");
    }
    //print double spaces inbetween pyramid sides
    printf("  ");
    //print left side hashes
    for (int j = 0; j < line; j++)
    {
      printf("#");
    }
    //move to next line
    printf("\n");
    line++;
  }
}
    

File: pset6/mario/mario.py
from cs50 import get_int

# ask user for pyramid height
# it should be a valid a integer between 1 and 8
while True:
    height = get_int("Height: ")
    if height in range(1, 9):
        break

# for each line
for i in range(1, height + 1):
    # print spaces
    print((height - i) * " ", end="")
    # print left side hashes
    print(i * "#", end="")
    # print double space in the middle
    print("  ", end="")
    # print right side hashes
    print(i * "#")